### PR TITLE
Validate enum namespace in file context are unique

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: wire-compiler/src/test/java/com/squareup/wire/WireCompilerErrorTest.kt [hunk 1]
```diff
@@ -119,6 +119,30 @@ class WireCompilerErrorTest {
...
   }
 
+  @Test
+  fun testEnumNamespaceFile() {
+    val e = assertFailsWith<SchemaException> {
+      compile("""
+          |package com.squareup.protos.test;
+          |
+          |enum Bar {
+          |  QUIX = 0;
```
Comment (swankjesse) [inline wire-compiler/src/test/java/com/squareup/wire/WireCompilerErrorTest.kt]: FWIW, `0` is a terrible value for an enum constant with protos

**Comment by swankjesse** [comment]
Protos are bad.

File: wire-compiler/src/test/java/com/squareup/wire/WireCompilerErrorTest.kt (hunk 1 above)
Comment (oldergod) [reply wire-compiler/src/test/java/com/squareup/wire/WireCompilerErrorTest.kt]: Because of proto3 absence of absence, right? Or even in proto2?

Comment (swankjesse) [reply wire-compiler/src/test/java/com/squareup/wire/WireCompilerErrorTest.kt]: Yeah.

**Pull request merged**